### PR TITLE
Simple Payments Button block - added example

### PR DESCRIPTION
**Pull request by MichaelArestad**
Master issue: #13510

Adds example to Simple Payments Button

#### Changes proposed in this Pull Request:

* Implements the ability to preview the block before inserting it.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* This is a new feature of Gutenberg we are moving to support.

#### Testing instructions:

* Go to the post editor
* Click the + to open a block inserter
* Hover over the simple payments button

#### Proposed changelog entry for your changes:

* We should probably add one changelog entry for all the blocks we add examples to. Something like: "Added previews to Jetpack blocks within the Gutenberg editor."

### CODE REVIEW

**Comment by simison** [comment]
The first version might be fine without a featured image if it turns out complicated.

**Comment by MichaelArestad** [comment]
@simison I agree. I just stubbed this one out for now and as @joanrho and I mock them up, we'll see what looks the best and go with that.

**Comment by joanrho** [comment]
Looks great, but no image (since i just sent that over to ya). Ping me again for review when you add the product image and I'll take another look.

File: extensions/blocks/simple-payments/index.js [hunk 1]
```diff
@@ -120,6 +120,16 @@ export const settings = {
 
 	save,
 
+	example: {
+		attributes: {
+			price: 25.0,
+			title: 'Jetpack t-shirt',
+			content:
+				'Take flight in ultimate comfort with ths stylish t-shirt featuring the Jetpack logo.',
```
Comment (simison) [inline extensions/blocks/simple-payments/index.js]: Remember to add `__()`s. This string will be a funky one for translators to see. :-D

Comment (MichaelArestad) [reply extensions/blocks/simple-payments/index.js]: Will do!

Comment (MichaelArestad) [reply extensions/blocks/simple-payments/index.js]: Added translations

File: extensions/blocks/simple-payments/index.js [hunk 2]
```diff
@@ -12,6 +12,11 @@ import edit from './edit';
 import save from './save';
 import { DEFAULT_CURRENCY } from './constants';
 
+/**
+ * Example image
+ */
+import simplePaymentsExample1 from './simple-payments__example-1.jpg';
```
Comment (jeherve) [inline extensions/blocks/simple-payments/index.js]: Is there a specific reason why you used 2 underscores here? You used only one for the Tiled Galleries examples:
`extensions/blocks/tiled-gallery/tiled-gallery_example-1.jpg`

Maybe we should stick to that to remain consistent?

**Review by jeherve — CHANGES_REQUESTED** [review]

File: extensions/blocks/simple-payments/index.js (hunk 2 above)
Comment (jeherve) [reply extensions/blocks/simple-payments/index.js]: Done in 3e7154719944a02dafa52162ceecc28b361d7408

**Review by jeherve — CHANGES_REQUESTED** [review]
While this works well when the block is already available on your site, it is problematic when you use the Free plan and want to preview the block with the upgrade nudge (cc @Automattic/luna):

image

To reproduce, switch to a free plan on your site and add the following to a functionality plugin on your site:
`add_filter( 'jetpack_block_editor_enable_upgrade_nudge', '__return_true' );` 

I think we'd need to hide the banner in block previews.

**Comment by simison** [comment]
Hmm, funky! I don't know a good way of hiding that apart from CSS. :-/

**Comment by MichaelArestad** [comment]
for now, i'll hide it with CSS

**Comment by simison** [comment]
Thanks. This nudge can appear also in Recurring payments block and there's another nudge for Stripe connections in recurring payments as well.

Hence it's prolly best to hide all those and future nudges with generic rule at here:

https://github.com/Automattic/jetpack/blob/f1c0d743ff628ad02f14c8d060a150ae003046bf/extensions/shared/components/block-nudge/style.scss

You can find upgrade nudge and Stripe nudges here if you'd prefer to hide those directly: https://github.com/Automattic/jetpack/tree/f1c0d743ff628ad02f14c8d060a150ae003046bf/extensions/shared/components

**Comment by MichaelArestad** [comment]
Add some styles I think resolve the issue, but waiting for JN to update to test.

**Comment by MichaelArestad** [comment]
but now it's no longer scaling correctly. Not sure what that's about.

image

**Comment by MichaelArestad** [comment]
@jeherve What scenario would a user see this upgrade nudge in a preview? I'm thinking of reverting my change because there's something up with the way it's scaling the preview now.

**Comment by jeherve** [comment]
> What scenario would a user see this upgrade nudge in a preview?

They'd see it when on a free plan, and when using the filter I mentioned above to activate the nudges (those nudges are not activated for everyone yet, hence the filter).

**Comment by simison** [comment]
The nudges are enabled for .com free site customers so once this code is merged and synced to wpcom, they'd see that nudge in previews.

**Comment by simison** [comment]
It could be possible to leave out preview when the nudge is visible. You'd have to use the same methods as are used in [`extensions/shared/register-jetpack-block.js`](https://github.com/Automattic/jetpack/blob/70ba68db209484a4676ce5691e8daef2e780ad2a/extensions/shared/register-jetpack-block.js) and it can get little involved.

In that file `wrapPaidBlock()` is the method that actually adds the nudge to the block.

**Comment by simison** [comment]
Oh, actually you could pull `example` out from the block if it requires a plan: 

https://github.com/Automattic/jetpack/blob/70ba68db209484a4676ce5691e8daef2e780ad2a/extensions/shared/register-jetpack-block.js#L47-L51

Something like this could work?
```js
example requiredPlan ? null : settings.example, 
```

(not sure if `example` has to be `false`,`null`, `undefined` or what to actually not render)

**Comment by MichaelArestad** [comment]
@simison Thanks for the suggestion. I tried just that and I _think_ it works. @jeherve when you get a chance, can you test this?

**Review by jeherve — APPROVED** [review]
This works well for me. Merging.

**Comment by ockham** [comment]
It's a bit unfortunate that we need to hide the block preview for the blocks that we might need it the most for -- the ones that users cannot fully use out-of-the-box before paying for an upgrade :sweat_smile:

**Comment by simison** [comment]
For sure — totally would be great to find a way for both Recurring payments and Simple payments. I think this is still a good mid-step just to get to today's code freeze.

**Pull request merged**